nix/hevm-tests/smt-checker.nix

```diff
@@ -0,0 +1,383 @@
+/*
+  This file explores the contracts in the test suite for the solc SMTChecker module
+  with hevm and compares the results with those found by the SMTChecker
+*/
+{ pkgs, solidity, solc, solver }:
+
+let
+
+  # --- binaries ---
+
+  awk = "${pkgs.gawk}/bin/awk";
+  bc = "${pkgs.bc}/bin/bc";
+  basename = "${pkgs.coreutils}/bin/basename";
+  cat = "${pkgs.coreutils}/bin/cat";
+  echo = "${pkgs.coreutils}/bin/echo";
+  grep = "${pkgs.gnugrep}/bin/grep";
+  hevm = "${pkgs.hevm}/bin/hevm";
+  jq = "${pkgs.jq}/bin/jq";
+  mkdir = "${pkgs.coreutils}/bin/mkdir";
+  mktemp = "${pkgs.coreutils}/bin/mktemp";
+  nix = "${pkgs.nix}/bin/nix";
+  rev = "${pkgs.utillinux}/bin/rev";
+  rm = "${pkgs.coreutils}/bin/rm";
+  sed = "${pkgs.gnused}/bin/sed";
+  tee = "${pkgs.coreutils}/bin/tee";
+  tr = "${pkgs.coreutils}/bin/tr";
+  timeout = "${pkgs.coreutils}/bin/timeout";
+
+  # --- test classification ---
+
+  ignored = [
+
+    # --- constructor arguments ---
+
+    "functions/constructor_hierarchy_3.sol"
+    "functions/constructor_hierarchy_4.sol"
+    "functions/constructor_hierarchy_diamond_2.sol"
+    "functions/constructor_hierarchy_diamond_3.sol"
+    "functions/constructor_hierarchy_diamond_empty_middle.sol"
+    "functions/constructor_hierarchy_diamond.sol"
+    "functions/constructor_hierarchy_empty_chain.sol"
+    "functions/constructor_hierarchy_empty_middle_no_invocation.sol"
+    "functions/constructor_hierarchy_empty_middle.sol"
+    "functions/constructor_hierarchy_mixed_chain_empty_base.sol"
+    "functions/constructor_hierarchy_mixed_chain.sol"
+    "functions/constructor_hierarchy_mixed_chain_local_vars.sol"
+    "functions/constructor_hierarchy_mixed_chain_with_params.sol"
+    "functions/constructor_hierarchy_mixed_chain_with_params_2.sol"
+    "functions/constructor_simple.sol"
+    "functions/constructor_state_value_inherited.sol"
+    "functions/constructor_state_value.sol"
+    "inheritance/constructor_hierarchy_mixed_chain_with_params.sol"
+    "inheritance/constructor_state_variable_init_chain_run_all.sol"
+    "inheritance/constructor_state_variable_init_chain_run_all_2.sol"
+
+    # --- unbounded looping ---
```

Comment:
We can add `--max-iterations n` and try some of these. It looks like `functions/functions_recursive.sol` and `functions/functions_recursive_indirect` should work, even for `n = 1` for example.